tests/e2e/common.go

```diff
@@ -294,3 +287,38 @@ func watchPods() {
 		}
 	}
 }
+
+// helper func to write `kubectl get -o yaml` output to file
+func writeToFile(fileName string, resource Output) error {
+
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	file, err := os.Create(fileName)
```

Comment:
We should probably avoid creating this again if we already find it in the map, right?